core/service/transaction/stake.ts

```diff
@@ -87,15 +86,16 @@ class TransactionStakeService implements IAssetService<IAssetStake>  {
         return (trs.asset.amount * config.constants.fees.froze) / TOTAL_PERCENTAGE;
     }
 
-    applyUnconfirmed(trs: Transaction<IAssetStake>): void {
-        const totalAmount: number = trs.fee + trs.asset.amount;
-        AccountRepo.updateBalanceByAddress(trs.senderAddress, -totalAmount);
+    applyUnconfirmed(trs: Transaction<IAssetStake>, sender: Account): void {
+        sender.actualBalance -= trs.asset.amount;
+        // TODO sender.stakes.push(new Stake())
+        // distribute airdrop
```

Comment:
for a stake, we will send direct reward?

Reply:
Yes, we will change airdrop and recipient account for airdrop here